core/src/main/scala/stryker4s/report/model/mutationTestReport.scala

```diff
@@ -0,0 +1,35 @@
+package stryker4s.report.model
+import io.circe.Encoder
+import stryker4s.report.model.MutantStatus.MutantStatus
+
+final case class MutationTestReport(schemaVersion: String,
+                                    thresholds: Thresholds,
+                                    files: Map[String, MutationTestResult]) {
+
+  private implicit val encoder: Encoder[MutantStatus] = Encoder.enumEncoder(MutantStatus)
+
+  def toJson: String = {
+    import io.circe.generic.auto._
+    import io.circe.syntax._
+    this.asJson.noSpaces
+  }
+}
+
+final case class MutationTestResult(source: String, mutants: Seq[MutantResult], language: String = "scala")
+
+final case class MutantResult(id: String,
+                              mutatorName: String,
+                              replacement: String,
+                              location: Location,
+                              status: MutantStatus)
+
+final case class Location(start: Position, end: Position)
+
+final case class Position(line: Int, column: Int)
+
+final case class Thresholds(high: Int, low: Int)
+
+object MutantStatus extends Enumeration {
```

Comment:
So ReporterType uses `sealed trait` and MutantStatus uses `extends Enumeration`. I'm thinking maybe we should keep things consistent, or at least have clear reasons for which implementation we use. :)

Is there a good reason to use Enumeration here?

Reply:
Oh it's probably because of the circe json encoder... 
https://github.com/stryker-mutator/stryker4s/blob/40ec153df66f2f32042f4a575a1130ccc62f0fa4/core/src/main/scala/stryker4s/report/model/mutationTestReport.scala#L9

Reply:
`ReporterType` uses a sealed trait with case objects because it is more scala-friendly than enumerations. In the `ConfigReaderImplicits` there's an [extra `ConfigReader`](https://github.com/stryker-mutator/stryker4s/blob/master/core/src/main/scala/stryker4s/config/implicits/ConfigReaderImplicits.scala#L17-L20) to read it in the way we want. I just didn't want to write an extra serializer for Circe.

Case objects get serialized to JSON/HOCON like this: `"killed": {}` instead of `"mutantStatus": "killed"`. This happens with both Circe (JSON) and PureConfig (HOCON). 

For serializing I thought an enum would be clearer than an extra serializer implementation for ADT's.

Reply:
I see, that makes sense.